torch/_inductor/autotuner/model.py

```diff
@@ -0,0 +1,801 @@
+import os
+import copy
+import pickle
+import logging
+import functools
+import itertools
+import dataclasses
+from typing import Dict, List, Tuple, NamedTuple
+
+import torch
+import torch._logging
+import numpy as np
+import re
+from torch import nn
+from torch._inductor.dependencies import StarDep, WeakDep
+from torch._inductor import config, dependencies, scheduler
+from torch._inductor.triton_heuristics import (
+    unique_configs,
+    pointwise_heuristic,
+    reduction_heuristic,
+    persistent_reduction_heuristic,
+)
+from torch._inductor.virtualized import V
+from torch._inductor.coordinate_descent_tuner import CoordescTuner, get_field, set_field
+from torch._inductor.utils import sympy_product, get_dtype_size
+
+log = logging.getLogger(__name__)
+
+
+### Raw data related
+class AutotunerRawData(NamedTuple):
+    io_deps: Tuple[
+        List[dependencies.Dep], List[dependencies.Dep], List[int], List[int], List[int]
+    ]
+    read_write: dependencies.ReadWrites
+    src_code: str
+    autotuner_dict: Dict
+
+    def __repr__(self):
+        return f"AutotunerRawData({self.io_deps}, {self.read_write}, {self.src_code}, {self.autotuner_dict})"
+
+
+def get_reads_writes(cur_scheduler, node, src_code):
```

Comment:
Would these be better as methods on the scheduler objects?

Reply:
It's only used here so I feel for now it's great to put it in this file.